src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -667,7 +667,9 @@ def _get_index(self):
             The union of all indexes across the partitions.
         """
         # TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset
-        return self._query_compiler.index
+        from snowflake.snowpark.modin.plugin.extensions.index import Index
+
+        return Index(data=self, convert_to_lazy=True)
```

Comment:
You should update it in qc.index right?

Reply:
based on offline discussion, currently updating the `qc.index` is not possible since `qc.index` is a property and we would like to pass the original df/series to Index. Therefore, perform the multiindex check in the dataframe layer and call `qc.index` when necessary.